server/rpc/yorkie_server.go

```diff
@@ -129,6 +129,11 @@ func (s *yorkieServer) AttachDocument(
 		return nil, err
 	}
 
+	// check DocumentKey is valid before it is attached
+	if err := pack.DocumentKey.Validate(); err != nil {
```

Comment:
Is document key validation needed per every attachment?
I guess that it is because `AttachDocuent()` does both  attachment and creation by DB upsert.

Reply:
That's a good comment.

I thought a lot about this part.

I also don't think it's appropriate to check every time in attach.

However, the operation of mongo and memory database is different.

In the case of mongo, upsert is one action, so we need to separate it into two actions.

I didn't make a clear decision on the state that should be detached, so I checked it before Attach.

I think this part needs more discussion.